pkg/service/storage/ucan/blob_accept.go

```diff
@@ -0,0 +1,70 @@
+package ucan
+
+import (
+	"context"
+
+	"github.com/storacha/go-libstoracha/capabilities/blob"
+	"github.com/storacha/go-ucanto/core/invocation"
+	"github.com/storacha/go-ucanto/core/receipt/fx"
+	"github.com/storacha/go-ucanto/core/result/failure"
+	"github.com/storacha/go-ucanto/principal"
+	"github.com/storacha/go-ucanto/server"
+	"github.com/storacha/go-ucanto/ucan"
+
+	"github.com/storacha/piri/pkg/pdp"
+	"github.com/storacha/piri/pkg/service/blobs"
+	"github.com/storacha/piri/pkg/service/claims"
+	blobhandler "github.com/storacha/piri/pkg/service/storage/handlers/blob"
+)
+
+type BlobAcceptService interface {
+	ID() principal.Signer
+	PDP() pdp.PDP
+	Blobs() blobs.Blobs
+	Claims() claims.Claims
+}
+
+func BlobAccept(storageService BlobAcceptService) server.Option {
```

Comment:
very minor preference nitpick, feel free to disregard: add a `With` prefix to the method name to make it a bit clearer it returns an option, a `Method` suffix (or even `ServiceMethod`, but that might be too verbose) because it returns, more specifically, a `ServiceMethod` option, and use a more generic name for the argument.

E.g.
```suggestion
func WithBlobAcceptMethod(service BlobAcceptService) server.Option {
```

and equivalent changes in the other functions.